src/controllers/user.controller.ts

```diff
@@ -0,0 +1,110 @@
+/* eslint-disable prettier/prettier */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/require-await */
+
+import UserService from '@src/services/user.service';
+import { Request, Response } from 'express';
+import { t } from 'i18next';
+import { comparePassword, hashPassword } from '@src/utils/passwordUtils';
+import { UserRoles } from '@src/enums/UserRoles.enum';
+
+export class UserController {
+  public static getRegister = async (req: Request, res: Response) => {
+    try {
+      res.render('pages/user/register', { pageTitle: t('action.register') });
+    } catch (error) {
+      req.flash('error_msg', t('error.registerFail'));
+      return res.render('error', {
+        message: t('error.loginFail'),
+        error: { status: 500, stack: error.stack },
+      });
+    }
+  };
+  public static getLogin = async (req: Request, res: Response) => {
+    try {
+      res.render('pages/user/login', { pageTitle: t('action.login') });
+    } catch (error) {
+      req.flash('error_msg', t('noPageLogin'));
+      return res.render('error', {
+        message: t('error.loginFail'),
+        error: { status: 500, stack: error.stack },
+      });
+    }
+  };
+  public static postRegister = async (req: Request, res: Response) => {
+    try {
+      const data = req.body;
+      const password = req.body.password;
+      // Mã hóa mật khẩu trước khi lưu
+      const hashedPassword = await hashPassword(password);
+      // Thay thế mật khẩu gốc bằng mật khẩu đã mã hóa
+      data.password = hashedPassword;
+      data.role = UserRoles.User;
+      const isValue = await UserService.create(data);
+      if (!isValue) {
+        req.flash('error_msg', t('error.createFail'));
+        return res.redirect('/register');
+      }
+      return res.redirect('/login');
```

Comment:
Không đặt hết logic vào try như thế này, chỉ đặt những logic có khả năng trả về exception thôi. Tương tự cho những chỗ khác
```suggestion
    const data = req.body;
    const password = req.body.password;
    
    try {
      // Mã hóa mật khẩu trước khi lưu
      const hashedPassword = await hashPassword(password);
      const isValue = await UserService.create({ ..data, password: hashedPassword, role: UserRoles.User });
      if (!isValue) {
        req.flash('error_msg', t('error.createFail'));
        return res.redirect('/register');
      }
      return res.redirect('/login');
```